xkbcommon/xkbcommon.h

```diff
@@ -1140,6 +1155,48 @@ xkb_level_index_t
 xkb_keymap_num_levels_for_key(struct xkb_keymap *keymap, xkb_keycode_t key,
                               xkb_layout_index_t layout);
 
+/**
+ * Retrieves every possible modifier mask that produces the specified
+ * shift level for a specific key and layout.
+ *
+ * This API is useful for inverse key transformation; ie. finding out
+ * which modifiers need to be active in order to be able to type the
+ * keysym(s) corresponding to the specific key code, layout and level.
+ *
+ * @warning It returns only up to masks_size modifier masks. If the
+ * buffer passed is too small, some of the possible modifier combinations
+ * will not be returned.
+ *
+ * @param[in] keymap      The keymap.
+ * @param[in] key         The keycode of the key.
+ * @param[in] layout      The layout for which to get modifiers.
+ * @param[in] level       The shift level in the layout for which to get the
+ * modifiers. This must be smaller than:
```

Comment:
```suggestion
 * modifiers. This should be smaller than:
```

Reply:
The doc text was copied from the doc for the nearby `xkb_keymap_key_get_syms_by_level` function. Should it not remain consistent?

Reply:
I pushed a commit to master which changes it there. It's allowed to try an out of range level (not an error), just not very sensible.